File: apps/frontend/src/types/triage.ts
export type ModelMode = "auto" | "online" | "offline";
export type SourceLanguage = "auto" | "en" | "fr" | "ar-SD";
export type TargetLanguage = "en" | "fr" | "ar-SD";

export type TriageRequest = {
  session_id?: string | null;
  transcript: string;
  source_language: SourceLanguage;
  target_language: TargetLanguage;
  model_mode: ModelMode;
  patient: Record<string, unknown>;
  measurements: Record<string, unknown>;
  context: Record<string, unknown>;
};

export type TriageResultData = {
  session_id: string;
  triage_color: string;
  classification: string;
  human_review_required: boolean;
  extracted_symptoms: Record<string, unknown>;
  missing_information: string[];
  tool_results: Array<Record<string, unknown>>;
  reasoning: string;
  recommendations: string[];
  translated_output: string;
  safety_flags: string[];
  citations: Array<Record<string, unknown>>;
  model: {
    mode?: string;
    name?: string;
    provider?: string;
    [key: string]: unknown;
  };
  errors: Array<Record<string, unknown>>;
};

File: apps/backend/schemas/triage.py
from typing import Any, Literal

from pydantic import BaseModel, Field


LanguageCode = Literal["auto", "en", "fr", "ar-SD"]
TargetLanguageCode = Literal["en", "fr", "ar-SD"]
ModelModePreference = Literal["auto", "online", "offline"]


class TriageRequest(BaseModel):
    session_id: str | None = None
    transcript: str = Field(min_length=1)
    source_language: LanguageCode = "auto"
    target_language: TargetLanguageCode = "en"
    model_mode: ModelModePreference = "auto"
    patient: dict[str, Any] = Field(default_factory=dict)
    measurements: dict[str, Any] = Field(default_factory=dict)
    context: dict[str, Any] = Field(default_factory=dict)


class TriageResultData(BaseModel):
    session_id: str
    triage_color: str
    classification: str
    human_review_required: bool
    extracted_symptoms: dict[str, Any]
    missing_information: list[str]
    tool_results: list[dict[str, Any]]
    reasoning: str
    recommendations: list[str]
    translated_output: str
    safety_flags: list[str]
    citations: list[dict[str, Any]]
    model: dict[str, Any]
    errors: list[dict[str, Any]] = Field(default_factory=list)
